src/Compilers/CSharp/Portable/BoundTree/Formatting.cs

```diff
@@ -82,17 +84,18 @@ public override object Display
 
 
                 builder.Append('(');
-                builder.Append(arguments[0].Display);
+                builder.Append("{0}");
 
-                for(int i = 1; i < arguments.Length; i++)
+                for (int i = 1; i < arguments.Length; i++)
                 {
-                    builder.Append(", ");
-                    builder.Append(arguments[i].Display);
+                    builder.Append(", {" + i + "}");
                 }
 
                 builder.Append(')');
 
-                return pooledBuilder.ToStringAndFree();
+                var format = pooledBuilder.ToStringAndFree();
+                var argumentDisplays = arguments.Select(a => a.Display).ToArray();
```

Comment:
>var argumentDisplays = arguments.Select(a => a.Display).ToArray(); [](start = 16, length = 66)

Instead of doing this, allocate an array of the right size for arguments and populate it along with the format string.
```
var argumentsDisplay = new object[arguments.Length];
...
argumentsDisplay[i] = arguments[i].Display;
...
FormattableStringFactory.Create(format, argumentsDisplay)
```

Reply:
Make sure to remove ```using System.Linq;``` if it is no longer needed.

---
In reply to: [172283897](https://github.com/dotnet/roslyn/pull/25209#discussion_r172283897) [](ancestors = 172283897)